.github/workflows/ci.yml

```
@@ -33,15 +33,32 @@ jobs:
         run: |
           pip install -r requirements.txt
           pip install -r docs/requirements.txt
+      
+      - name: Install lcov
+        run: |
+          sudo apt-get update
+          sudo apt-get install -y lcov
 
       - name: Build package
         run: |
-          python scripts/build/install.py
-
+          CXXFLAGS=--coverage CFLAGS=--coverage python scripts/build/install.py
+# coverage tests     
       - name: Run tests
         run: |
           python -m pytest --doctest-modules --cov=./ --cov-report=xml -s
 
+      - name: Capture Coverage Data with lcov
+        run: |
+          cd build
+          sudo apt-get update
+          sudo apt-get install -y lcov
+          lcov --capture --directory . --output-file coverage.info
```

```suggestion
          lcov --capture --directory . --output-file coverage.info --no-external
```

If you check the logs, then you will C++'s internal headers are also covered. We only want PyDataStructs C++ source files.